sympy/functions/special/gamma_functions.py

```diff
@@ -1015,3 +1016,92 @@ def trigamma(x):
     .. [3] http://functions.wolfram.com/GammaBetaErf/PolyGamma2/
     """
     return polygamma(1, x)
+
+###############################################################################
+##################### COMPLETE MULTIVARIATE GAMMA FUNCTION ####################
+###############################################################################
+
+class multivariate_gamma(Function):
+    r"""
+    The multivariate gamma function is a generalization of the gamma function i.e,
+
+    .. math::
+        \Gamma_p(z) = \pi^{p(p-1)/4}\prod_{k=1}^p \Gamma[z + (1 - k)/2].
+
+    Special case, multivariate_gamma(x, 1) = gamma(x)
+
+    Parameters
+    ==========
+
+    p: order or dimension of the multivariate gamma function
+
+    Examples
+    ========
+
+    >>> from sympy import S, I, pi, oo, gamma, multivariate_gamma
+    >>> from sympy import Symbol
+    >>> x = Symbol('x')
+    >>> p = Symbol('p', positive=True, integer=True)
+
+    >>> multivariate_gamma(x, p)
+    pi**(p*(p - 1)/4)*Product(gamma(-_k/2 + x + 1/2), (_k, 1, p))
+
+    Several special values are known:
+    >>> multivariate_gamma(1, 1)
+    1
+    >>> multivariate_gamma(4, 1)
+    6
+    >>> multivariate_gamma(S(3)/2, 1)
+    sqrt(pi)/2
+
+    Writing multivariate_gamma in terms of gamma function
+    >>> multivariate_gamma(x, 1)
+    gamma(x)
+
+    >>> multivariate_gamma(x, 2)
+    sqrt(pi)*gamma(x)*gamma(x - 1/2)
+
+    >>> multivariate_gamma(x, 3)
+    pi**(3/2)*gamma(x)*gamma(x - 1)*gamma(x - 1/2)
+
+    See Also
+    ========
+
+    gamma, lowergamma, uppergamma, polygamma, loggamma, digamma, trigamma
+    sympy.functions.special.beta_functions.beta
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Multivariate_gamma_function
+    """
+    unbranched = True
+
+    def fdiff(self, argindex=2):
+        from sympy import Sum
+        if argindex == 2:
+            x, p = self.args[0], self.args[1]
+            k = Dummy('k')
+            return self.func(x, p)*Sum(polygamma(0, x + (1 - k)/2), (k, 1, p))
+        else:
+            raise ArgumentIndexError(self, argindex)
+
+    @classmethod
+    def eval(cls, x, p):
+        from sympy import Product
+        x, p = list(map(sympify, (x, p)))
+        if p.is_positive == False or p.is_integer == False:
+            raise ValueError('Order parameter must be positive integer.')
+        if p.is_positive == None or p.is_integer == None:
+            p = Symbol('p', integer=True, positive=True)
```

Comment:
Careful with redefinition of `p`:
```
In [1]: multivariate_gamma(n, m)                                                
Out[1]: 
 p⋅(p - 1)    p                   
 ───────── ─┬────┬─               
     4      │    │   ⎛  k       1⎞
π         ⋅ │    │  Γ⎜- ─ + n + ─⎟
            │    │   ⎝  2       2⎠
            │    │                
            k = 1                 

In [2]: _.atoms(Symbol)                                                         
Out[2]: {k, n, p}
```

Reply:
Done. Thanks!

Reply:
It should be able to handle generic expressions as arguments as well.
```py
>>> var('x y')
>>> multivariate_gamma(x, y + 1)
...
AttributeError: 'Add' object has no attribute 'name'
```